src/lib/bootloader/device_map.rb

```diff
@@ -2,73 +2,76 @@
 
 require "bootloader/udev_mapping"
 
+require "yast2/target_file"
+require "cfa/grub2/device_map"
+
 module Bootloader
   # Class representing grub device map structure
   class DeviceMap
+    extend Forwardable
     include Yast::Logger
 
-    def initialize(mapping = {})
+    def_delegators :@model, :grub_device_for, :system_device_for, :grub_devices,
+      :add_mapping, :remove_mapping
+
+    def initialize
       # lazy load to avoid circular dependencies
       Yast.import "Arch"
       Yast.import "BootStorage"
       Yast.import "Mode"
       Yast.import "Storage"
-      @mapping = mapping
+      @model = CFA::Grub2::DeviceMap.new
     end
 
-    def to_hash
-      @mapping.dup
+    def to_s
+      "Device Map: #{@model.grub_devices}"
     end
 
-    def to_s
-      "Device Map: #{@mapping.inspect}"
+    def read
+      @model.load
+    end
+
+    def write
+      @model.save
     end
 
     def size
-      @mapping.size
+      @model.grub_devices.size
     end
 
     def empty?
-      @mapping.empty?
+      size == 0
     end
 
-    def contain_disk?(disk)
-      @mapping.include?(disk) ||
-        @mapping.include?(::Bootloader::UdevMapping.to_mountby_device(disk))
+    def clear_mapping
+      @model.grub_devices.each do |grub_dev|
+        @model.remove_mapping(grub_dev)
+      end
     end
 
-    def disks_order
-      disks = @mapping.select { |_k, v| v.start_with?("hd") }.keys
+    def contain_disk?(disk)
+      disk = @model.grub_device_for(disk) ||
+        @model.grub_device_for(::Bootloader::UdevMapping.to_mountby_device(disk))
 
-      disks.sort_by { |d| @mapping[d][2..-1].to_i }
+      !disk.nil?
     end
 
-    # Function remap device map to device name (/dev/sda)
-    # or to label (ufo_disk)
-    # @return [Hash{String => String}] new device map
-    def remapped_hash
-      if !Yast::Arch.ppc
-        return to_hash if Yast::Storage.GetDefaultMountBy == :label
-      end
-
-      # convert device names in device map to the device names by device or label
-      remapped = @mapping.map do |k, v|
-        [UdevMapping.to_kernel_device(k), v]
-      end
+    def disks_order
+      grub_devices = @model.grub_devices
+      grub_devices.select! { |d| d.start_with?("hd") }
+      grub_devices.sort_by! { |dev| dev[2..-1].to_i }
 
-      Hash[remapped]
+      grub_devices.map { |grub_dev| @model.system_device_for(grub_dev) }
```

Comment:
np: I would use a single-letter name for `dev` and `grub_dev` variables (according to openSUSE Ruby style guide). Moreover, it would be consistent with the `#select!` block.